src/utilities/IOManager.cpp

```diff
@@ -17,6 +17,7 @@ bool IOManager::init(bool isPrimaryAI) {
 
     auto worldCallback = std::bind(&IOManager::handleState, this, std::placeholders::_1);
     this->worldSubscriber = std::make_unique<rtt::net::WorldSubscriber>(worldCallback);
+    simulationConfigurationPublisher = std::make_unique<rtt::net::SimulationConfigurationPublisher>();
```

Comment:
You never initialized simulationConfigurationPublisher. You use it here as a variable that already exists, but you have not added this to IOManager.h.

Reply:
Ah yes, you are right. How does this even work at all then? `simulationConfigurationPublisher` should at this point be local to `IOManager::init()`, but it still exists in `IOManager::sendSimulationConfiguration`?